### PR TITLE
feat(core): add alternativeTranslations to vocabulary generation

### PR DESCRIPTION
**Pull request by ceolinwill**
## Summary

- Add `alternativeTranslations` field to AI vocabulary schema, DB model, and data pipeline
- Create `getDistractorWords()` utility in `@zoonk/core` for filtering semantically equivalent words from distractor pools
- Update evals with new criterion and Portuguese greetings/farewells test case
- Add integration tests for saving and updating alternativeTranslations

## Test plan

- [x] Unit tests for `getDistractorWords()` (12 tests covering all filtering rules, edge cases, shuffling)
- [x] Integration tests for saving and updating `alternativeTranslations` in workflow
- [x] All existing tests updated and passing
- [x] `pnpm typecheck`, `pnpm knip`, `pnpm test`, builds all pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
## Summary by cubic
Adds alternativeTranslations to vocabulary words across generation, storage, and API so synonymous answers don’t appear as distractors. Also adds a distractor selection utility, updates prompts/evals, and includes a DB migration.

- **New Features**
  - Add alternativeTranslations: string[] to Word model and AI vocabulary schema; save on create/update.
  - Expose alternativeTranslations in getActivity, getLessonWords, and prepareActivityData.
  - Add getDistractorWords utility to filter same/equivalent translations (case-insensitive, bidirectional) and overlapping alternatives; shuffle; unit tests.
  - Update vocabulary prompt with rules/examples; add Portuguese greetings/farewells eval; integration tests added; existing tests updated.
  - Remove stray tasks/ISSUES.md.

- **Migration**
  - Add words.alternative_translations (TEXT[] DEFAULT []); no backfill needed.

<sup>Written for commit 974d0ab42c5dcafe549de77d9b394471a7991ee0. Summary will update on new commits.</sup>

### CODE REVIEW

File: packages/core/src/player/get-distractor-words.test.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  test("shuffles the result", () => {
+    const correct = makeWord("1", "target");
+    const words = [
+      correct,
+      ...Array.from({ length: 20 }, (_, index) =>
+        makeWord(String(index + 2), `word-${String(index)}`),
+      ),
+    ];
+
+    const results = Array.from({ length: 10 }, () =>
```
[the remark this answers is not in the document]
Comment (ceolinwill) [reply packages/core/src/player/get-distractor-words.test.ts:116]: This is a false positive. With 20 items shuffled into groups of 10, the probability of 10 consecutive identical orderings is roughly 1 in 10^150. This test will never flake. Making it deterministic via mocking would test the mock, not the actual shuffling behavior.
